transmart-core-db/src/main/groovy/org/transmartproject/db/ontology/AbstractQuerySpecifyingType.groovy

```diff
@@ -86,6 +86,12 @@ abstract class AbstractQuerySpecifyingType implements MetadataSelectQuerySpecifi
         PatientDimension.findAllByIdInList(patientIdList)
     }
 
+    protected int countPatients(OntologyTerm term) {
```

Comment:
I understand why this function is here, because `getPatients` is as well, but they can both better be moved to a service, e.g., the `PatientSetQueryBuilderService`. Then the querying code is grouped together in the service and domain classes are not cluttered with retrieval code.